talk/morelanguage/stl.tex

```diff
@@ -163,22 +165,40 @@
 \begin{frame}[fragile]
   \frametitlecpp[98]{STL and functors}
   \begin{cppcode}
-    // Finds the first element in a list that lies in
-    // the range from 1 to 10.
-    list<int> L;
+    // Finds the first element in a list between 1 and 10.
+    list<int> l;
     ...
-    list<int>::iterator in_range =
-      find_if(L.begin(), L.end(),
+    list<int>::iterator it =
+      find_if(l.begin(), l.end(),
               compose2(logical_and<bool>(),
                        bind2nd(greater_equal<int>(), 1),
                        bind2nd(less_equal<int>(), 10)));
 
     // Computes sin(x)/(x + DBL_MIN) for elements of a range.
     transform(first, last, first,
-              compose2(divides<double>(),
+              compose2(divides<double>(), // non-standard
```

Comment:
I don't know what to do here. `compose2` is not standard, but available in some STL implementations. The alternative is `std::bind`, which is C++11.